jina/clients/mixin.py

```diff
@@ -167,3 +170,23 @@ async def post(
     search = partialmethod(post, '/search')
     update = partialmethod(post, '/update')
     delete = partialmethod(post, '/delete')
+
+
+def _wrap_on_error(on_error):
+
+    num_args = len(signature(on_error).parameters)
+    if num_args == 1:
+        warnings.warn(
+            "on_error callback taking only the response parameters is deprecated. Please add one parameter "
+            "to handle additional optional Exception as well",
+            DeprecationWarning,
+        )
+
+        @wraps(on_error)
```

Comment:
I am not sure u need this extra decoration? you are already wrapping it by returning `_fn`

Reply:
Ok I see, this way it is more transparent!